pkg/vex/vex.go

```diff
@@ -115,6 +131,28 @@ func determineStatus(packageVersion string) vex.Status {
 	return vex.StatusFixed
 }
 
-func generateDocumentID(packageName string) string {
-	return fmt.Sprintf("vex-%s-%s", packageName, uuid.New())
+// generateDocumentID generate a deterministic document ID based
+// on the configuration data contents
+func generateDocumentID(configs []*build.Configuration) (string, error) {
+	hashes := []string{}
+	for _, c := range configs {
+		data, err := yaml.Marshal(c)
+		if err != nil {
+			return "", fmt.Errorf("marshaling melange configuration")
+		}
+		h := sha256.New()
+		if _, err := h.Write(data); err != nil {
+			return "", fmt.Errorf("hashing melange configuration")
+		}
+		hashes = append(hashes, fmt.Sprintf("%x", h.Sum(nil)))
+	}
+
+	sort.Strings(hashes)
+	h := sha256.New()
+	if _, err := h.Write([]byte(strings.Join(hashes, ":"))); err != nil {
+		return "", fmt.Errorf("hashing config files")
+	}
+
+	// One hash to rule them all
+	return fmt.Sprintf("vex-%s", fmt.Sprintf("%x", h.Sum(nil))), nil
```

Comment:
I noticed the errors returned in this function don't include the `err` value (e.g. via `%w`), is that intentional?

Reply:
Funny, I don't know why I forgot all the wrappings there. I've added them